sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -198,41 +198,6 @@ class ConvertHelper : public ReturnHelper {
     return PI_SUCCESS;
   }
 
-  // Convert the array (0-terminated) using a conversion map
-  template <typename TypeUR, typename TypePI>
-  pi_result convertArray(std::function<TypePI(TypeUR)> Func) {

```

Comment:
please explain what exactly was wrong with this helper and we can't keep it (fixed)?

Reply:
The convertArray() function only works for the specific case where:

  - The array is 0 terminated
  - There is no need to change the layout of the array when converting
  
The only cases where this function was being used was for UR_DEVICE_INFO_PARTITION_TYPE and UR_DEVICE_INFO_SUPPORTED_PARTITIONS. The first case is no longer suitable because the layout of the array has to change when converting. In addition, both cases are not 0 terminated. It is possible to modify the function for the second case (i.e. make it work with 0 terminated arrays). However, since it is the only case where the function is still usable, I thought that it was unnecessary to keep it and that removing this extra generic function would make the conversions happening in the UR_DEVICE_INFO_SUPPORTED_PARTITIONS case easier to understand.

Reply:
So , there are still 2 users, and there can be mor in future. I suggest keeping this utility and adjust it to be even more generic, i.e. allow shrinking conversions.